eks/modules.tf

```diff
@@ -0,0 +1,32 @@
+module "network" {
+    source = "./modules/network"
+}
+
+module "eks" {
+    source = "./modules/eks"
+
+    cluster_name  = "${module.network.generated-cluster-name}"
+    vpc-id = "${module.network.vpc_id}"
+    app_subnet_ids = "${module.network.app_subnet_ids}"
+    workstation_cidr_block = "${var.workstation_cidr_block}"
+    aws-network-dependency-id = "${module.network.aws-network-dependency}"
+    keypair_name = "${var.keypair_name}"
+    eks_version = "${var.eks_version}"
+    cluster_labels = "${var.cluster_labels}"
+}
+
+module "services" {
+    source = "./modules/services"
+
+    aws-eks-cluster-name = "${module.eks.aws-eks-cluster-name}"
+    aws-eks-cluster-endpoint = "${module.eks.aws-eks-cluster-endpoint}"
+    aws-eks-cluster-certificate-authority-data = "${module.eks.aws-eks-cluster-certificate-authority-data}"
+    worker-arn = "${module.eks.aws-node-arn}"
+    force-eks-dependency-id = "${module.eks.force-eks-dependency-id}"
+    region = "${var.region}"
+    hosted_zone_name = "${var.hosted_zone_name}"
+    hosted_zone_id = "${module.services.hosted_zone_id}"
```

Comment:
I'm not sure I follow this...we are calling services module passing in a variable `hosted_zone_id` which is referring to an output from the same child module with the same name? Why do you not move the data source for the zone into the network module to get the output and then use it here as an input?

Reply:
You are correct `hosted_zone_id` does not need to be passed to services module explicitly. I have made changes to infer it directly using `"${data.aws_route53_zone.selected.zone_id}"` at all required places within services module. This might have been there as a consequence of moving config across modules.